src/drivers/dw/dma.c

```diff
@@ -1433,70 +1315,52 @@ static void dw_dma_irq_handler(void *data)
 	if (!status_intr)
 		return;
 
-	tracev_dwdma("dw-dma: %d IRQ status 0x%x", dma->plat_data.id,
-		     status_intr);
-
-	/* get the source of our IRQ. */
-	status_block = dw_read(dma, DW_STATUS_BLOCK);
-	status_tfr = dw_read(dma, DW_STATUS_TFR);
+	tracev_dwdma("dw_dma_irq_handler(): dma %d IRQ status 0x%x",
+		     dma->plat_data.id, status_intr);
 
-	/* clear interrupts */
-	dw_write(dma, DW_CLEAR_BLOCK, status_block);
-	dw_write(dma, DW_CLEAR_TFR, status_tfr);
+	/* get the source of our IRQ and clear it */
+#if CONFIG_HW_LLI
+	status_src = dw_read(dma, DW_STATUS_BLOCK);
+	dw_write(dma, DW_CLEAR_BLOCK, status_src);
+#else
+	status_src = dw_read(dma, DW_STATUS_TFR);
+	dw_write(dma, DW_CLEAR_TFR, status_src);
+#endif
 
 	/* TODO: handle errors, just clear them atm */
 	status_err = dw_read(dma, DW_STATUS_ERR);
-	dw_write(dma, DW_CLEAR_ERR, status_err);
-	if (status_err)
-		trace_dwdma_error("dw-dma: %d error 0x%x", dma->plat_data.id,
-				  status_err);
+	if (status_err) {
+		trace_dwdma_error("dw_dma_irq_handler() error: dma %d status "
+				  "error 0x%x", dma->plat_data.id, status_err);
+		dw_write(dma, DW_CLEAR_ERR, status_err);
+	}
 
 	/* clear platform and DSP interrupt */
-	pmask = status_block | status_tfr | status_err;
-	platform_interrupt_clear(dma_irq(dma, cpu_get_id()), pmask);
-
-	/* confirm IRQ cleared */
-	status_block_new = dw_read(dma, DW_STATUS_BLOCK);
-	if (status_block_new) {
-		trace_dwdma_error("dw-dma: %d status block 0x%x not cleared",
-				  dma->plat_data.id, status_block_new);
-	}
+	platform_interrupt_clear(dma_irq(dma, cpu_get_id()),
```

Comment:
@tlauda shouldn't also this instance have been replaced with dma_chan_irq()? To simply reproduce the driver behaviour before the patch, you'd have to make this call conditional on AGGREGATE. BTW, this anyway will be heavily modified with the cascading IRQ patch series.

Reply:
@lyakh You are right, I've missed it, but fortunately, it doesn't change anything. platform_interrupt_clear does nothing on the APL, which is the only platform without aggregation.